src/vs/workbench/contrib/terminal/common/terminalEnvironment.ts

```diff
@@ -128,9 +128,16 @@ export function getCwd(shell: IShellLaunchConfig, userHome: string, root?: Uri,
 
 	// TODO: Handle non-existent customCwd
 	if (!shell.ignoreConfigurationCwd && customCwd) {
-		if (path.isAbsolute(customCwd)) {
+		if (configurationResolverService) {
+			try {
+				cwd = configurationResolverService.resolve(lastActiveWorkspace, customCwd);
+			} catch {
+				cwd = undefined;
```

Comment:
This can be removed

Reply:
If I don't have the try catch then the terminal hangs if you place something like ${fileDirName} in with no file open

Reply:
It hangs? It should show an error notification because of https://github.com/microsoft/vscode/pull/76743, we don't want it to fail silently.

Reply:
Yeah it just appears to be a blank terminal with nothing in it, no error or anything. By having the try catch it then gives an error `unknown shell cwd /Users/.../${fileDirName}`

Reply:
Improved to fail hard and log to console
![image](https://user-images.githubusercontent.com/2193314/62087117-75af7080-b215-11e9-808e-c2254cba6dbf.png)